crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -145,12 +145,21 @@ pub trait RouteAndFill: StateWrite + Sized {
 
         let max_delta_1: Amount = MAX_RESERVE_AMOUNT.into();
 
+        let mut execution_circuit_breaker = ExecutionCircuitBreaker::new();
```

Comment:
If we make `route_and_fill` take an `ExecutionCircuitBreaker`, and make the upper bound for path search and fills a parameter of that struct, this will let us craft test a variety of test vectors for the circuit breaking code. We can keep the constants for the production code.

Reply:
That's already the case. I guess I would ditch the construct hiding those details though.